pkg/collector/process.go

```diff
@@ -137,19 +137,19 @@ func (cp *CollectingProcess) CloseMsgChan() {
 	close(cp.messageChan)
 }
 
-func (cp *CollectingProcess) GetNumOfRecordsReceived() uint64 {
+func (cp *CollectingProcess) GetNumRecordsReceived() int64 {
 	cp.mutex.RLock()
 	defer cp.mutex.RUnlock()
-	return cp.numOfRecordsReceived
+	return int64(cp.numOfRecordsReceived)
```

Comment:
Why do we need to use `int64` here? Is there a possibility for negative records?